### PR TITLE
docs: add Chatting Guide

### PR DESCRIPTION
**Pull request by hieu-jan**
PR Description: 
This pull request implements the guide documentation for Chatting section
- [x] #1136: Overall guide for chatting
- [x] #1129: Guide for Starting Thread
- [x] #1132: Guide for Manage Chat History

Check-list based on new feedback: In-progress: Not Ready for Review
- Manage Chat History
- [x] Entire GIF in new format (Light mode, clear view)
- [x] Adding missing section managing thread via folders
- [x] Adding missing image/ gif
- Starting a thread (Temporary remove in this PR)

### CODE REVIEW

File: docs/docs/guides/03-chatting/02-upload-docs.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+    Jan AI,
+    Jan,
+    ChatGPT alternative,
+    local AI,
+    private AI,
+    conversational AI,
+    no-subscription fee,
+    large language model,
+  ]
 ---
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/02-upload-docs.md:16]: Insert a `info` section: similiar to: https://jan.ai/developer/build-assistant/your-first-assistant/

File: docs/docs/guides/03-chatting/03-upload-images.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+  [
+    Jan AI,
+    Jan,
+    ChatGPT alternative,
+    local AI,
+    private AI,
+    conversational AI,
+    no-subscription fee,
+    large language model,
+  ]
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/03-upload-images.md:15]: Insert info section, similar to https://jan.ai/developer/build-assistant/your-first-assistant/

File: docs/docs/guides/03-chatting/01-start-thread.md [hunk 3]
```diff
@@ -1,12 +1,40 @@
...
-  At thread level
-  Globally, as default
-Choosing a model
-Customizing model params (thread level)
-Customizing engine params
+A thread is the conversations between an assistant and a user. This tutorial will walk you through the process of starting a thread on Jan.
+
+### Setting Thread Title
+
+A thread title acts as a name for your thread. It appears on the left side of the chat window, helping you easily navigate through your interactions. If you do not set a thread title, Jan will set it to "New Thread" as default.
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/01-start-thread.md:24]: Missing screenshot

File: docs/docs/guides/03-chatting/01-start-thread.md [hunk 4]
```diff
@@ -1,12 +1,40 @@
...
+
+### Setting Assistant Instructions
+
+On Jan, you can set assistant instructions that guide the responses of your assistant. These instructions are set at the conversation level. Globally, as default, assistant instructions will apply to all conversations in this thread unless you change it at the conversation level. In one thread, you can change the assistant instructions for each conversation. If you do not set assistant instructions, Jan will set it to empty as default.
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/01-start-thread.md:28]: This is confusing. 

Break into two secitons please: 

1. Setting Assistant Instructions (at the thread level)
2. 1. Setting Default Assistant Instructions (at the global level)

Then include screenshots.

File: docs/docs/guides/03-chatting/01-start-thread.md [hunk 5]
```diff
@@ -1,12 +1,40 @@
...
+
+### Choosing a Model
+
+Jan offers a variety of models to suit your conversational needs. Each model has its unique characteristics and capabilities. You can choose a model that best fits your needs via the hub and download your preferred model. Then, you can set the model for your thread.
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/01-start-thread.md:32]: Missing the following info: 

1. Currently, you can only change models in a new thread. After a thread is initiated, you can't change models.

File: docs/docs/guides/03-chatting/01-start-thread.md [hunk 6]
```diff
@@ -1,12 +1,40 @@
...
+
+### Customizing Model Params
+
+You can customize the model params for your thread e.g. setting a customized value of max_tokens value for your thread, etc.
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/01-start-thread.md:36]: Missing: 

1. Changing model configs at the thread level
2. Changing model configs at the assistant level
3. Changing model configs at the global/default level

Include screenshots for users - this feature has been a source of great confusion (our fault)

image

File: docs/docs/guides/03-chatting/04-manage-history.md [hunk 7]
```diff
@@ -1,3 +1,43 @@
...
+![viewing-chat-history](./assets/manage-history-1.gif)
+
+## Clean Thread
+
+To streamline your conservation view, click on the three dots (⋮) on the thread you want to clean, then select `Clean Thread`. It will remove all messages from the thread. It is useful if you want to keep the thread settings, but want to remove the messages from the chat window.
+
+<br></br>
+![clean-thread](./assets/manage-history-2.gif)
+
+## Delete Thread
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/04-manage-history.md:38]: Missing section: 

## Managing Threads via Folders
1. Indicate to users they can directly access both thread history and thread configs via their folders. 
2. Show them how to `reveal in finder` `view as json`
3. etc.

File: docs/docs/guides/03-chatting/README.md [hunk 8]
```diff
@@ -1,3 +1,24 @@
...
+    local AI,
+    private AI,
+    conversational AI,
+    no-subscription fee,
+    large language model,
+    chatting,
+  ]
 ---
+
+This guide is designed to help you maximize your experience with Jan, covering everything from starting engaging threads to managing your chat history effectively.
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/README.md:19]: Thanks for this. Very helpful to have this.

File: docs/docs/guides/03-chatting/README.md [hunk 9]
```diff
@@ -1,3 +1,24 @@
...
+
+- [Start a thread](start-thread)
+- [Upload docs](upload-docs)
+- [Upload images](upload-images)
+- [Manage chat history](manage-chat-history)
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/README.md:24]: I can't leave a comment directly in the gifs, but I think it's more helpful to record gifs in lightmode. Our darkmode is kinda hard to read at the moment.

**Review by 0xSage — CHANGES_REQUESTED** [review]
A few nits. Can we try to get this PR in asap.

File: docs/docs/guides/03-chatting/README.md (hunk 9 above)
Comment (0xSage) [reply docs/docs/guides/03-chatting/README.md:24]: Non blocking for this PR. Just pls note for future.

File: docs/docs/guides/03-chatting/01-start-thread.md [hunk 10]
```diff
@@ -1,12 +1,47 @@
...
-Customizing engine params
+A thread is the conversations between an assistant and a user. This tutorial will walk you through the process of starting a thread on Jan.
+
+<br></br>
+
+![start-thread-gif](./assets/start-thread.gif)
+
+### Setting Thread Title
+
+A thread title acts as a name for your thread. It appears on the left side of the chat window, helping you easily navigate through your interactions. If you do not set a thread title, Jan will set it to 'New Thread' as default.
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/01-start-thread.md:28]: Hey, let's treat our users with respect. 

Assume that users (like us) are lazy but smart. 

Meaning, we don't need to explain simple things like "A thread title is a title of a thread."

I'd like this entire PR to be 50% more concise. 

Please pass your copy through Jan assistant iwth the instructions "Help me rephrase these user guides to be more concise and helpful."

File: docs/docs/guides/03-chatting/01-start-thread.md (hunk 5 above)
Comment (0xSage) [reply docs/docs/guides/03-chatting/01-start-thread.md:39]: Again, please split into 2 sections, with a screen recording for both

File: docs/docs/guides/03-chatting/04-manage-history.md [hunk 11]
```diff
@@ -1,3 +1,43 @@
...
+    local AI,
+    private AI,
+    conversational AI,
+    no-subscription fee,
+    large language model,
+    manage-chat-history,
+  ]
 ---
+
+Jan offers a convenient and private way to interact with a conversational AI locally on your computer. This guide will walk you through how to manage your chat history with Jan, ensuring your interactions remain private and organized.
```
Comment (0xSage) [inline docs/docs/guides/03-chatting/04-manage-history.md:19]: First sentence is not related to this page. 

Isn't the main message more of: Jan persists your app usage locally on your filesystem, so your data never leaves your computer. This guide will walk you through how to direct manage your application files...

**Review by 0xSage — CHANGES_REQUESTED** [review]

**Comment by 0xSage** [comment]
1. I think the chatting section is missing a major user guide teaching users that **they need to download a model from the Hub first**. The ordering of these pages need to be improved. 

2. Also, a lot of the screenshots in the /chatting pages say "You don't have a local model yet".

**Comment by hieu-jan** [comment]
Note: Temporary hide starting thread and taking all current issues in the other PR. There are some changes in current UI of starting thread. Will need to re-documentation this part and record based on new UI.

**Comment by hieu-jan** [comment]
Note: Temporary hide starting thread, not ready docs for models and taking all current issues into consideration in the other PR. There are some changes in current UI of starting thread. Will need to re-documentation this part and record based on new UI.

**Review by 0xSage — APPROVED** [review]

**Pull request merged**